File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.swift
import Foundation

func solution(_ s:String) -> Int {
    
    var stack: [Character] = []
    
    for char in s {
        if stack.isEmpty {
            stack.append(char)
        } else if stack.last! == char {
            stack.removeLast()
        } else {
            stack.append(char)
        }
    }

    return stack.isEmpty ? 1 : 0
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
# 문자열 길이 100만


def solution(s):
    
    stack = []
    
    for char in s:
        if not stack:
            stack.append(char)
        elif stack[-1] == char:
            stack.pop()
        else:
            stack.append(char)
    
    return 0 if stack else 1